engine/schema/src/main/java/com/cloud/vm/dao/UserVmDetailsDaoImpl.java

```diff
@@ -22,13 +22,45 @@
 import org.apache.cloudstack.resourcedetail.ResourceDetailsDaoBase;
 
 import com.cloud.vm.UserVmDetailVO;
+import com.cloud.vm.VMInstanceVO;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import javax.inject.Inject;
 
 @Component
 public class UserVmDetailsDaoImpl extends ResourceDetailsDaoBase<UserVmDetailVO> implements UserVmDetailsDao {
 
+    @Inject
+    protected UserVmDetailsDao _detailsDao;
+
     @Override
     public void addDetail(long resourceId, String key, String value, boolean display) {
         super.addDetail(new UserVmDetailVO(resourceId, key, value, display));
     }
 
+    @Override
+    public void saveDetails(VMInstanceVO vm) {
+        saveDetails(vm, new ArrayList<String>());
+    }
+
+    @Override
+    public void saveDetails(VMInstanceVO vm, List<String> hiddenDetails) {
```

Comment:
@DaanHoogland 
it is called by 
```
public void saveDetails(VMInstanceVO vm) {
```
which is called by
```
userVmDetailsDao.saveDetails
```

Reply:
yes, so it is not needed as `saveDetails(VMInstanceVO vm)` passes a constant empty `ArrayList`.

Reply:
This is a copy of the code used in UserVmManagerImpl for user vms where the method is used. Think that could be moved to the UserVmDetailsDaoImpl too.
Would this move be wanted in this PR too? if so it is not ready. Otherwise the PR is ready imo.

Reply:
Yes @BartJM , that would definitely make sense. It deduplicates code. But as the code uses the `VMInstanceVO` to get the `vm.id`, it could create a cyclic dependency. I don't think it is much effort so if you could give it a short try, that would be much appreciated.

Reply:
I updated the other code to use the UserVmDetailsDaoImp too. This method is now used from somewhere else. Was not too much effort and did not encounter a cyclic dependency the way I deduplicated.